app/renderer/components/navigation/urlBar.js

```diff
@@ -357,10 +357,13 @@ class UrlBar extends React.Component {
   componentDidUpdate (prevProps) {
     // this.urlInput is not initialized in titleMode
     if (this.urlInput) {
+      const pdfjsEnabled = getSetting(settings.PDFJS_ENABLED)
       if (this.props.activeFrameKey !== prevProps.activeFrameKey) {
         this.keyPressed = false
         // The user just changed tabs
-        this.setValue(this.props.locationValue)
+        this.setValue(this.props.locationValue !== 'about:blank'
+          ? this.props.locationValue
+          : UrlUtil.getDisplayLocation(this.props.location, pdfjsEnabled))
```

Comment:
I don't understand what this does.

Only in the case that`this.props.locationValue` is `about:blank` does it go to `UrlUtil.getDisplayLocation(this.props.location, pdfjsEnabled))` and `UrlUtil.getDisplayLocation('about:blank', pdfjsEnabled))` actually returns about:blank.